### PR TITLE
Convert CS lessons to be Javascript specific

### PR DESCRIPTION
**Pull request by CouchofTomato**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**
Javascript does not currently have a CS course

**2. This PR:**

* Copies the Ruby CS course files into the Javascript directory and then edits references to anything Ruby specific to be Javascript specific.

**3. Additional Information:**

### CODE REVIEW

File: javascript/computer_science/project_binary_search_trees.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+1. Write an `insert` and `delete` functions which accepts a value to insert/delete (you'll have to deal with several cases for delete such as when a node has children or not). If you need additional resources, check out these two articles on [inserting](https://www.geeksforgeeks.org/binary-search-tree-set-1-search-and-insertion/?ref=lbp) and [deleting](https://www.geeksforgeeks.org/binary-search-tree-set-2-delete/?ref=lbp), or [this video](https://youtu.be/wcIRPqTR3Kc) with several visual examples.
+
+1. Write a `find` function which accepts a value and returns the node with the given value.
+
+1. Write a `level_order` function which accepts another function as a parameter. `level_order` should traverse the tree in breadth-first level order and yield each node to the provided function. This function can be implemented using either iteration or recursion (try implementing both!). The method should return an array of values if no function is given. **Tip:** You will want to use an array acting as a queue to keep track of all the child nodes that you have yet to traverse and to add new ones to the list (as you saw in the [video](https://www.youtube.com/watch?v=86g8jAQug04)).
```
Comment (01zulfi) [inline javascript/computer_science/project_binary_search_trees.md:22]: Yield each node to the provided function sounds weird for JS. Wording suggestion:
```suggestion
1. Write a `levelOrder` function which accepts another function as a parameter. `levelOrder` should traverse the tree in breadth-first level order and provide each node as the argument to the provided function. This function can be implemented using either iteration or recursion (try implementing both!). The method should return an array of values if no function is given. **Tip:** You will want to use an array acting as a queue to keep track of all the child nodes that you have yet to traverse and to add new ones to the list (as you saw in the [video](https://www.youtube.com/watch?v=86g8jAQug04)).
```

File: javascript/computer_science/project_knights_travails.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  * `knight_moves([3,3],[0,0]) == [[3,3],[1,2],[0,0]]`
+
+<div class="lesson-content__panel" markdown="1">
+1. Put together a script that creates a game board and a knight.
+2. Treat all possible moves the knight could make as children in a tree.  Don't allow any moves to go off the board.
+2. Decide which search algorithm is best to use for this case.  Hint: one of them could be a potentially infinite series.
+3. Use the chosen search algorithm to find the shortest path between the starting square (or node) and the ending square.  Output what that full path looks like, e.g.:
+
+~~~bash
+  > knight_moves([3,3],[4,3])
```
Comment (01zulfi) [inline javascript/computer_science/project_knights_travails.md:29]:
```suggestion
  > knightMoves([3,3],[4,3])
```

**Review by 01zulfi — CHANGES_REQUESTED** [review]
This is great 🔥 

I've left a review below, mostly it's just convention shift from Ruby to JS 😆 

Additionally, where do you envision this section going in the JavaScript course?

File: javascript/computer_science/recursive_methods.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  1. Read [this recursion resource](https://javascript.info/recursion) for a good intro to recursion.
+  2. Watch [this explanation of Recursion](https://www.youtube.com/watch?v=6oDQaB2one8) by Web Dev Simplified.
+  3. Watch this [Video on Recursion](https://www.youtube.com/watch?v=mz6tAJMVmfM) from CS50.
+  4. Read the ["Implementation Issues" section of the wiki article](http://en.wikipedia.org/wiki/Divide_and_conquer_algorithm#Implementation_issues) to get an overview of some of the limitations of recursion.
+</div>
+
+### Test Yourself
+
+<div class="lesson-content__panel" markdown="1">
+  1. Solve each of the questions from this [Code Quiz](http://www.codequizzes.com/computer-science/beginner/recursion) on Recursion. You are not expected to answer them from memory at this point.
```
[the remark this answers is not in the document]
Comment (CouchofTomato) [reply javascript/computer_science/recursive_methods.md:35]: What do you think about https://www.codingame.com/playgrounds/5422/js-interview-prep-recursion as an alternative

**Review by KevinMulhern — APPROVED** [review]
Looks great to me mate @CouchofTomato 👑

File: javascript/computer_science/recursive_methods.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (01zulfi) [reply javascript/computer_science/recursive_methods.md:35]: Some of the problems seem to be a little on the tough side, but I like it. 🚀

**Review by 01zulfi — APPROVED** [review]
Approved, although minor nitpicks 🚀

**Pull request merged**